### PR TITLE
Add options to endpoint registration

### PR DESCRIPTION
**Pull request by jcorbin**
Encapsulates streaming handler interface

r @kriskowal @Raynos

### CODE REVIEW

File: node/endpoint-handler.js [hunk 1]
```diff
@@ -41,11 +41,16 @@ inherits(TChannelEndpointHandler, EventEmitter);
...
+TChannelEndpointHandler.prototype.register = function register(name, options, handler) {
     var self = this;
+    if (typeof options === 'function') {
+        handler = options;
+        options = {};
+    }
     if (typeof handler !== 'function') {
         throw errors.InvalidHandlerError();
     }
+    if (options.streamed) handler.canStream = true;
```
Comment (kriskowal) [inline node/endpoint-handler.js]: It seems unclear to me whether the handler is accepting a streamed request or providing a streamed response.

**Comment by kriskowal** [comment]
:ok:

File: node/endpoint-handler.js (hunk 1 above)
Comment (jcorbin) [inline node/endpoint-handler.js]: It means you can handle streaming incoming.  Whether you send a streamed response is down to how you call `buildResponse(options)`.

**Comment by Raynos** [comment]
lgtm.

**Pull request merged**